server.js

```diff
@@ -1,29 +1,98 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import dotenv from 'dotenv'
+import listEndpoints from 'express-list-endpoints'
+
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true, useFindAndModify: false })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 9000
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const thoughtSchema = new mongoose.Schema({
+ message: {
+   type: String,
+   required: [true, 'Message is required'],
+   minlength: [5, 'Minimum length is 5 characters'],
+   maxlength: [140, 'Maximum length is 140 characters']
+ },
+ hearts: {
+  type: Number,
+  default: 0
+ },
+ createdAt: {
+   type: Date,
+   default: Date.now
+ }
+})
+
+const Thought = mongoose.model('Thought', thoughtSchema)
+
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
+
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({
+      message: 'Service unavailable'
+    })
```

Comment:
I guess this is  middleware handling if server is unavailable and sending error message? Very nice adding this to the app. 🎉